src/cursor/cur_std.c

```diff
@@ -1052,6 +1052,15 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     } else
         WT_ERR_NOTFOUND_OK(ret, false);
 
+    /* Set the prefix search near flag. */
+    if ((ret = __wt_config_getones(session, config, "prefix_key", &cval)) == 0) {
```

Comment:
I feel like there should be restrictions on when the prefix search can be used. Probably only for simple schema type row store tables. Could you add a check and error message here?

Reply:
Will do.

Reply:
Also custom collators should be disallowed.